x-pack/auditbeat/module/system/socket/state.go

```diff
@@ -804,6 +925,15 @@ func (f *flow) toEvent(final bool) (ev mb.Event, err error) {
 				metricset["euid"] = f.process.euid
 				metricset["egid"] = f.process.egid
 			}
+
+			if resolver != nil {
+				if domain, found := resolver.ResolveIP(f.pid, f.local.addr.IP); found {
```

Comment:
So does this enrichment happen at the end of the "connection"?

I'm thinking about the case of a long running connection where the it does an initial DNS query then keeps the connection open for hours. So it might be better to do the enrichment with domain as soon as we have the endpoints of the socket available?

Reply:
Ouch, that was an oversight on my side. Thanks for catching. I've refactored it so that reverse DNS is stored by process.